birdyboard.py

```diff
@@ -0,0 +1,505 @@
+                self.users_menu_next_step()
+            finally:
+                # TODO: fix this. it doesn't work.
+                try:
+                    if int(next_step)-1 >= 0:
+                        user_index = int(next_step)-1
+                        self.enter_password(user_index)
+                    else:
+                        print("your number is not in the list of users.")
+                        self.users_menu_next_step()
+                except IndexError:
+                    print("your number is not in the list of users.")
+                    self.users_menu_next_step()
+                finally:
+                    pass
+
+    def enter_password(self, user_index):
+        """
+        runs as part of users_menu_next_step. receives the index number error-handled by users_menu_next_step, receives user-input for a password and error handles that, and passes both into check_password to verify. if check_password returns true, takes you to the logged-in top-level menu. If it returns false, asks you again. Also gives you the option to go directly to the 'create new user' menu.
+
+        Argument: integer value that represents the index of a user in the self.users list.
+        """
+        print("'x' to exit.\n'b' to go back.\n'n' to create new user.")
+
+        password_try = input("password: ")
+        if password_try == "b":
+            print("going back.")
+            self.view_users_menu()
+            self.users_menu_next_step()
+        elif password_try == "x":
+            print("goodbye.")
+            exit()
+        elif password_try == "n":
+            print("new user menu.")
+            self.create_a_user_menu()
+        else:
+            verify = self.check_password(user_index, password_try)
+            if verify is True:
+                print(self.logged_in_menu)
+                self.logged_in_menu_next_step()
+            else:
+                print("incorrect Password.")
+                self.enter_password()
+
+    def check_password(self, user_index, password_try):
+        """
+        runs as part of enter_password. receives the user index number passed through from users_menu_next_step and the password entered in enter_password, and checks to see if they are equal. if they are, assigns the user and returns 'true' to enter_password for continuation.
+
+        Arguments: 1. integer value that represents the index of a user in the self.users list, and 2. string to check agains the password for the user with the matching index.
+
+        """
+        current_user = self.users[user_index]
+        if password_try == current_user["password"]:
+            print("logging in to " + current_user["user_name"] + ".")
+            self.user_name = current_user["user_name"]
+            return True
+        else:
+            return False
+
+# ########################################
+# ######## VIEW ALL CHIRPS ###############
+# ########################################
+
```

Comment:
Your "View all chirps" and "View full chirp thread" methods are good candidates for rolling into their own class.